pkg/resourceconverter/resource_converter.go

```diff
@@ -22,53 +23,87 @@ type TagField struct {
 	Value string `yaml:"value"`
 }
 
+func (f TagField) IsZero() bool {
+	return f.Name == ""
+}
+
 type ReflectionConverter struct {
-	ResourceType string
-	TagField     TagField
-	IdField      string
-	Region       string
+	ResourceType   string
+	TagField       TagField
+	IdField        string
+	DisplayIdField string
+	Region         string
 }
 
 func (rc *ReflectionConverter) ToResource(ctx context.Context, x any, tags model.Tags) (model.Resource, error) {
-	t := reflect.TypeOf(x)
-
-	// get the id field
-	var id string
-	for i := 0; i < t.NumField(); i++ {
-		field := t.Field(i)
-		name := field.Name
-		if name == rc.IdField {
-			fieldPtrRef := reflect.ValueOf(x).FieldByName(name)
-			fieldRef := reflect.Indirect(fieldPtrRef)
-			if !fieldRef.IsZero() {
-				id = fmt.Sprintf("%v", fieldRef.Interface())
-			}
-			break
-		}
+	resource := model.Resource{
+		Region: rc.Region,
+		Type:   rc.ResourceType,
 	}
 
-	if id == "" {
+	resource.Id = rc.findId(rc.IdField, x)
+	if resource.Id == "" {
 		return model.Resource{}, fmt.Errorf("could not find id field '%v' for type '%v", rc.IdField, rc.ResourceType)
 	}
 
+	if err := rc.loadDisplayId(x, &resource); err != nil {
+		return model.Resource{}, err
+	}
+
 	// generate tags field
-	if tags == nil {
+	if !rc.TagField.IsZero() {
```

Comment:
you can also implement `TagField.Exists()` to make the code more readable

Reply:
I did a quick look around, and it seemed like `IsZero` is a common method (`time.Time.IsZero` and `reflect.Value.IsZero` for two examples in the stdlib)